### PR TITLE
Release 2.3.0

### PR DESCRIPTION
**Pull request by uvvpavel**
Part of XVC-110.

### CODE REVIEW

File: test/asr/check_asr.sh [hunk 1]
```diff
@@ -21,6 +21,18 @@ help()
    echo "   h     Print this Help."
 }
 
+# Writes the XS3 TestMode register in the JTAG domain to reboot the device into JTAG-boot mode. 
+# Unsets the 'reboot' bit to release the chip from reset and waits for connection.
+# Will be Fixed in XTC > 15.3.0 (Bugzilla ID 18895).
```
Comment (mbanth) [inline test/asr/check_asr.sh:26]: Given that this release updates the repository to use tools 15.3.0, is this function still needed?

This comment applies to multiple instances of this function which appear in different .sh files.

File: doc/programming_guide/asrc/overview.rst [hunk 2]
```diff
@@ -72,7 +92,7 @@ Download the main repo and submodules using:
 Building the app
 ================
 
-First install and source the XTC version: 15.2.1 tools. The output should be
+First install and source the XTC version: |TOOLS_VERSION| tools. For example with version 15.2.1, the output should be
```
[the remark this answers is not in the document]
Comment (uvvpavel) [reply doc/programming_guide/asrc/overview.rst:95]: I didn't write this bit of documentation and not sure why it's been written in that way. It's one of those things that the person needs to remember when updating XTC tools. Very manual and not for a lot of purpose. I found it quite last minute and by luck, so probably not good for anyone new coming to work on this repo.
I've been battling myself between removing it altogether or leaving it as a reference output, left it this way...

File: doc/programming_guide/asr/deploying/linux_macos.rst [hunk 3]
```diff
@@ -27,15 +27,16 @@ Run the following commands in the root folder to build the host application usin
...
 
-Before running the host application, you may need to add the location of ``xscope_endpoint.so`` to your ``LD_LIBRARY_PATH`` environment variable.  This environment variable will be set if you run the host application in the XTC Tools command-line environment.  For more information see `Configuring the command-line environment <https://www.xmos.ai/documentation/XM-014363-PC-LATEST/html/tools-guide/install-configure/getting-started.html>`__.
+Before running the host application, you may need to add the location of ``xscope_endpoint.so`` to your ``LD_LIBRARY_PATH`` environment variable.  This environment variable will be set if you run the host application in the XTC Tools command-line environment.  For more information see `Configuring the command-line environment <https://xmos.com/xtc-install-guide>`__.
 
 Building the Firmware
 =====================
 
-Run the following commands in the root folder to build the firmware:
+After having your python environment activated, run the following commands in the root folder to build the firmware:
```
[the remark this answers is not in the document]
Comment (uvvpavel) [reply doc/programming_guide/asr/deploying/linux_macos.rst:35]: So this is copied from: fwk_voice, it's basically hiding
pyenv local 3.10
python -m venv .venv
. .venv/bin/activate
But not all users work with python that way so we didn't want to prescribe it. Also saves a lot of lines that are not directly related to the build process

File: test/asr/check_asr.sh (hunk 1 above)
Comment (mbanth) [reply test/asr/check_asr.sh:26]: Sorry, my mistake.  I read the comment as 'Will be fixed in XTC => 15.3.0'.

File: examples/ffd/src/i2c_reg_handling.c [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    uint8_t * data_p = *data;
+    uint8_t reg_addr = data_p[0];
+    uint8_t reg_value = 0xFF;
+    if (reg_addr == appconfINTENT_I2C_REG_ADDRESS) {
+        reg_value = last_asr_result->id;
+    }
+    data_p[0] = reg_value;
+    rtos_printf("Read from register 0x%02X value 0x%02X\n", reg_addr, reg_value);
+#endif
+    return 1;
```
[the remark this answers is not in the document]
Comment (uvvpavel) [reply examples/ffd/src/i2c_reg_handling.c:26]: If the given define is 0, then the rtos_i2c_slave thread will [never start](https://github.com/xmos/sln_voice/blob/develop/examples/ffd/bsp_config/XK_VOICE_L71/platform/platform_start.c#L56) and will never call this API. As far as grep shows me, it's defined to 1 in [this cmake](https://github.com/xmos/sln_voice/blob/develop/examples/ffd/ffd_i2s_input_cyberon.cmake#L83) with no comments

File: doc/programming_guide/ffd/software_desc/intent_engine.rst [hunk 5]
```diff
@@ -1,12 +1,12 @@
...
 
-#################
-src/intent_engine
-#################
+#########################
+modules/asr/intent_engine
+#########################
 
-This folder contains the intent engine module for the FFD application.
+This folder contains the intent engine module for the  and FFVA application.
```
[the remark this answers is not in the document]
Comment (mbanth) [reply doc/programming_guide/ffd/software_desc/intent_engine.rst:7]: Thanks for fixing this.

File: examples/ffd/src/i2c_reg_handling.c (hunk 4 above)
[the remark this answers is not in the document]
Comment (mbanth) [reply examples/ffd/src/i2c_reg_handling.c:26]: Thanks for checking.  I'm happy to have it left as it is.

File: doc/programming_guide/asrc/overview.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (mbanth) [reply doc/programming_guide/asrc/overview.rst:95]: Fair enough.  Let's leave it alone.

File: doc/programming_guide/asr/deploying/linux_macos.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (mbanth) [reply doc/programming_guide/asr/deploying/linux_macos.rst:35]: Okay.

**Review by mbanth — APPROVED** [review]

**Pull request merged**